main.tf

```
@@ -42,15 +42,15 @@ locals {
   create_group_enabled = module.this.enabled && var.ses_group_enabled
   create_user_enabled  = module.this.enabled && var.ses_user_enabled
 
-  ses_group_name = coalesce(var.ses_group_name, module.this.id)
+  ses_group_name = try(coalesce(var.ses_group_name, module.this.id), null)
```

This resolves the error that occurs when `enabled` is set to `false`.

---

I feel we should do an if true check to make the code more readable. This is unique to when the module is enabled only.
```suggestion
  ses_group_name = local. create_group_enabled ? coalesce(var.ses_group_name, module.this.id) :  null
```